pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -283,14 +292,7 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
         # dump states as a checkpoint dictionary object
         checkpoint = self.on_save(checkpoint)
         if self.is_global_zero:
```

Comment:
To maintain current behaviour I've kept the global zero rank check here, do you think that's sufficient for now @ananthsub?